src/test/java/uk/gov/pay/adminusers/resources/ServiceResourceUpdateMerchantDetailsTest.java

```diff
@@ -0,0 +1,105 @@
+package uk.gov.pay.adminusers.resources;
+
+import com.google.common.collect.ImmutableMap;
+import org.junit.Test;
+import uk.gov.pay.adminusers.model.Service;
+
+import java.util.Map;
+
+import static com.jayway.restassured.http.ContentType.JSON;
+import static java.lang.String.format;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.Matchers.hasItems;
+import static org.hamcrest.Matchers.hasKey;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.core.Is.is;
+import static uk.gov.pay.adminusers.app.util.RandomIdGenerator.randomInt;
+import static uk.gov.pay.adminusers.app.util.RandomIdGenerator.randomUuid;
+
+public class ServiceResourceUpdateMerchantDetailsTest extends IntegrationTest {
+
+    @Test
+    public void shouldSuccess_whenUpdatingMerchantDetails() throws Exception {
+        String serviceExternalId = randomUuid();
+        Service service = Service.from(randomInt(), serviceExternalId, "existing-name");
+        databaseHelper.addService(service, randomInt().toString());
+        Map<String, Object> payload = ImmutableMap.<String, Object>builder()
+                .put("name", "somename")
+                .put("address_line1", "line1")
+                .put("address_line2", "line2")
+                .put("address_city", "city")
+                .put("address_country", "country")
+                .put("address_postcode", "postcode").build();
+
+        givenSetup()
+                .when()
+                .contentType(JSON)
+                .accept(JSON)
+                .body(mapper.writeValueAsString(payload))
+                .put(format(SERVICE_RESOURCE, serviceExternalId) + "/merchant-details")
+                .then()
+                .statusCode(200)
+                .body("merchant_details.name", is("somename"))
+                .body("merchant_details.address_line1", is("line1"))
+                .body("merchant_details.address_line2", is("line2"))
+                .body("merchant_details.address_city", is("city"))
+                .body("merchant_details.address_country", is("country"))
+                .body("merchant_details.address_postcode", is("postcode"));
+    }
+
+    @Test
+    public void shouldSuccess_whenUpdatingMerchantDetails_withoutOptionalFields() throws Exception {
```

Comment:
Could you have a chat with @heathd about this? I think we discussed the possibility to rebalance the testing pyramid. So for example the sad cases being tests in an more unit test fashion. Not sure about this tho because maybe then we need to refactor more like:
- Resources exclusively deserialise the request, call the service and return a success response afterwards.
- Services handle the request object, do whatever is intended to do and throw an specific related exception and this exception is handle in a ExceptionHandler, responding according to the exception thrown.
- This way the service can be widely unit tested and the resource is only responsible to handle the happy case. So then integration tests can comply with a better balanced testing pyramid.

Reply:
Great idea @rauligs ! Addressed that in [1c1cac1] using DropWizard `ExceptionMapper` interface